pkg/morph/event/container/notifications.go

```diff
@@ -126,3 +128,60 @@ func RestoreEACLChanged(notifEvent *state.ContainedNotificationEvent) (event.Eve
 
 	return res, nil
 }
+
+// Transfer is a NEP-11 transfer event thrown by Container contract.
+type Transfer struct {
+	event.Event
+	Container cid.ID
+	From      util.Uint160
+	To        util.Uint160
+}
+
+// RestoreTransfer restores [Transfer] event from the notification one.
+func RestoreTransfer(notifEvent *state.ContainedNotificationEvent) (event.Event, error) {
+	items, err := getItemsFromNotification(notifEvent, 4)
+	if err != nil {
+		return nil, err
+	}
+
+	from, err := restoreItemValue(items, 0, "from", stackitem.ByteArrayT, client.BytesFromStackItem)
+	if err != nil {
+		return nil, err
+	}
+	to, err := restoreItemValue(items, 1, "to", stackitem.ByteArrayT, client.BytesFromStackItem)
+	if err != nil {
+		return nil, err
+	}
+	amount, err := restoreItemValue(items, 2, "amount", stackitem.IntegerT, client.IntFromStackItem)
+	if err != nil {
+		return nil, err
+	}
+	id, err := restoreItemValue(items, 3, "id", stackitem.ByteArrayT, client.BytesFromStackItem)
+	if err != nil {
+		return nil, err
+	}
+
+	var res Transfer
+
+	if amount != 1 {
+		return nil, fmt.Errorf("unexpected amount instead of 1: %d", amount)
+	}
+	if len(from) == 0 && len(to) == 0 {
+		return nil, errors.New("both 'from' and 'to' are empty")
+	}
+	if len(from) > 0 {
+		if res.From, err = util.Uint160DecodeBytesBE(from); err != nil {
+			return nil, fmt.Errorf("decode 'from' item: %w", err)
+		}
+	}
+	if len(to) > 0 {
+		if res.To, err = util.Uint160DecodeBytesBE(to); err != nil {
```

Comment:
Looks like `util.Uint160{}` and NULL become the same here. But that's the case for https://github.com/nspcc-dev/neo-go/blob/39f8079b7e58cc8dff570cb121e061a74ed784c0/pkg/core/blockchain.go#L2267-L2277 as well (CC @AnnaShaleva) even though `util.Uint160{}` is a correct NEO address that can be used in `to` (not likely to be used in `from` since it's hard to have a witness for it). Likely not much we can improve now, although for node `user.ID` is non-zero with `util.Uint160{}` hash used, so this can be leveraged.

Reply:
https://github.com/neo-project/proposals/issues/212